src/history/hs_cursor.c

```diff
@@ -151,21 +154,24 @@ __wt_hs_cursor_position(WT_SESSION_IMPL *session, WT_CURSOR *cursor, uint32_t bt
   const WT_ITEM *key, wt_timestamp_t timestamp, WT_ITEM *user_srch_key)
 {
     WT_DECL_RET;
-    WT_WITH_TXN_ISOLATION(session, WT_ISO_READ_UNCOMMITTED,
-      ret = __hs_cursor_position_int(session, cursor, btree_id, key, timestamp, user_srch_key));
+
+    WT_WITH_BTREE(session, CUR2BT(cursor),
+      WT_WITH_TXN_ISOLATION(session, WT_ISO_READ_UNCOMMITTED,
+        ret = __hs_cursor_position_int(session, cursor, btree_id, key, timestamp, user_srch_key)));
     return (ret);
 }
 
 /*
- * __wt_hs_find_upd --
- *     Scan the history store for a record the btree cursor wants to position on. Create an update
- *     for the record and return to the caller. The caller may choose to optionally allow prepared
- *     updates to be returned regardless of whether prepare is being ignored globally. Otherwise, a
- *     prepare conflict will be returned upon reading a prepared update.
+ * __wt_hs_find_upd_int --
+ *     Internal helper to scan the history store for a record the btree cursor wants to position on.
+ *     Create an update for the record and return to the caller. The caller may choose to optionally
+ *     allow prepared updates to be returned regardless of whether prepare is being ignored
+ *     globally. Otherwise, a prepare conflict will be returned upon reading a prepared update.
  */
-int
-__wt_hs_find_upd(WT_SESSION_IMPL *session, WT_ITEM *key, const char *value_format, uint64_t recno,
-  WT_UPDATE_VALUE *upd_value, bool allow_prepare, WT_ITEM *on_disk_buf, WT_TIME_WINDOW *on_disk_tw)
+static int
+__wt_hs_find_upd_int(WT_SESSION_IMPL *session, uint32_t btree_id, WT_ITEM *key,
```

Comment:
As this is an internal function I think it should just be `__hs_find_upd_int`.

Reply:
Fixed.